clients/src/main/java/org/apache/kafka/common/metrics/KafkaMetric.java

```diff
@@ -20,6 +20,37 @@
 import org.apache.kafka.common.MetricName;
 import org.apache.kafka.common.utils.Time;
 
+/**
+ * An implementation of {@link Metric} interface.
+ * <p>
+ * A KafkaMetric is a named metric for monitoring purpose. The metric value can be a {@link Measurable} or a {@link Gauge}.
+ * <pre>
+ * <b>metricName</b> The name of the metric
+ * <b>lock</b> A lock used for reading the metric value in case of race condition
+ * <b>time</b> The POSIX time in milliseconds the metric is being taken
+ * <b>metricValueProvider</b> The metric collecting implementation that implements {@link MetricValueProvider}
+ * <b>config</b> The metric configuration which is a {@link MetricConfig}
+ * </pre>
```

Comment:
This is already described in constructor, so I think we don't need it. WDYT?

Reply:
Yeah since we don't expect users to create KafkaMetric objects I don't think we need this.

Reply:
thanks for the suggestion! let me remove that